daemon/daemon.go

```diff
@@ -1221,16 +1221,16 @@ func (daemon *Daemon) verifyHostConfig(hostConfig *runconfig.HostConfig) ([]stri
 }
 
 func (daemon *Daemon) setHostConfig(container *Container, hostConfig *runconfig.HostConfig) error {
-	if err := daemon.registerMountPoints(container, hostConfig); err != nil {
-		return err
-	}
-
 	container.Lock()
 	defer container.Unlock()
 	if err := parseSecurityOpt(container, hostConfig); err != nil {
 		return err
 	}
 
+	if err := daemon.registerMountPoints(container, hostConfig); err != nil {
```

Comment:
this was outside the lock in purpose actually. We didn't want to block the container when volumes are created to avoid blocking docker ps due to external connections. Is there a reason to move it back here?

/cc @LK4D4